libpromises/dbm_lmdb.c

```diff
@@ -1355,7 +1355,7 @@ bool DBPrivWriteCursorEntry(
     {
         MDB_val curkey;
         curkey.mv_data = cursor->curkv;
-        curkey.mv_size = sizeof(cursor->curkv);
+        curkey.mv_size = strlen(cursor->curkv) + 1;
```

Comment:
`cursor->curkv` is `void *` and it's not guaranteed to be `\0`-terminated in LMDB, AFAICT.